From 1a271580719059034b965229f545fbbaec8b27ac Mon Sep 17 00:00:00 2001
From: Lewis Hanly <Lewis.Hanly@microchip.com>
Date: Tue, 25 Feb 2020 13:04:47 +0000
Subject: [PATCH] Microchip-Aloevera-DTS-Support

Adding Polarfire SoC DTS support for the following:
AXI-PCIE Host
MSS-GPIO/I2C/SPI/qSPI/CAN-UIO/USB


Signed-off-by: Lewis Hanly <Lewis.Hanly@microchip.com>
---
 arch/riscv/boot/dts/sifive/fu540-c000.dtsi | 89 ++++++++++++++++++++++
 1 file changed, 89 insertions(+)

diff --git a/arch/riscv/boot/dts/sifive/fu540-c000.dtsi b/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
index a2e3d54e830c..2d9fafafae95 100644
--- a/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
+++ b/arch/riscv/boot/dts/sifive/fu540-c000.dtsi
@@ -268,6 +268,95 @@
 			interrupts = <1 2 3>;
 			reg = <0x0 0x2010000 0x0 0x1000>;
 		};
+		pci@2030000000 {
+			#address-cells = <3>;
+			#size-cells = <2>;
+			compatible = "microchip,pf-axi-pcie-host-1.0";
+			device_type = "pci";
+			bus-range = <0x01 0x7f>;
+			interrupt-map = <0 0 0 1 &pcie_intc 1>,
+                                        <0 0 0 2 &pcie_intc 2>,
+                                        <0 0 0 3 &pcie_intc 3>,
+                                        <0 0 0 4 &pcie_intc 4>;
+			interrupt-map-mask = <0 0 0 7>;
+			interrupt-parent = <&plic0>;
+			#interrupt-cells = <1>;
+                        ranges = <0x3000000 0x0 0x40000000 0x0 0x40000000 0x0 0x20000000>;
+                        reg = <0x20 0x30000000 0x0 0x4000000 0x20 0x0 0x0 0x100000>;
+                        reg-names = "control", "apb";
+			pcie_intc: interrupt-controller {
+				#address-cells = <0>;
+				#interrupt-cells = <1>;
+				interrupt-controller;
+			};
+		};
+                L56: i2c@2000100000 {
+                        #address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "microsemi,ms-pf-mss-i2c";
+			reg = <0x20 0x00100000 0x0 0x1000>;
+			interrupt-parent = <&plic0>;
+			interrupts = <32>;
+			clocks = <&prci PRCI_CLK_TLCLK>;
+			clock-frequency = <400000>;
+                };
+		L57: gpio@2000103000 {
+			compatible = "microsemi,ms-pf-mss-gpio";
+			interrupt-parent = <&plic0>;
+			interrupts = <7 7 7 7 7 7 7>;
+			gpio-controller;
+			reg = <0x20 0x00103000 0x0 0x1000>;
+			reg-names = "control";
+			#gpio-cells = <2>;
+			#status = "okay";
+		};
+		L55: can@2000105000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "microsemi,ms-pf-mss-can-uio";
+			reg = <0x20 0x00105000 0x0 0x1000>;
+			interrupt-parent = <&plic0>;
+			clocks = <&prci PRCI_CLK_TLCLK>;
+			interrupts = <32>;
+		};
+		L54: spi@2000107000 {
+			compatible = "microsemi,ms-pf-mss-spi";
+			interrupt-parent = <&plic0>;
+			interrupts = <32>;
+			reg = <0x20 0x00107000 0 0x1000>;
+			clocks = <&prci PRCI_CLK_TLCLK>;
+			spi-max-frequency = <25000000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			num-cs = <8>;
+			status = "okay";
+		};
+		L58: qspi@2000400000 {
+			compatible = "microsemi,ms-pf-mss-qspi";
+			interrupt-parent = <&plic0>;
+			interrupts = <32>;
+			reg = <0x20 0x00400000 0 0x1000>;
+			clocks = <&prci PRCI_CLK_TLCLK>;
+			spi-max-frequency = <25000000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			num-cs = <8>;
+			status = "okay";
+		};
+		usb@2000200000 {
+			compatible = "microsemi,ms-pf-usb-host";
+			reg = <0x00000020 0x00200000 0x00000000 0x00001000>;
+			reg-names = "mc","control";
+			interrupt-parent = <0xb>;
+			interrupts = <0x20 0x20>;
+			interrupt-names = "mc","dma";
+			dr_mode = "host";
+			multipoint = <1>;
+			num-eps = <8>;
+			ram-bits = <12>;
+			power = <500>;
+			status = "ok";
+		};
 
 	};
 };
